backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java

```diff
@@ -126,31 +127,34 @@ public void updateOrgIdByRecordIdAndType(
 	/**
 	 * @return a map with one entry for each given {@link ExternalReferenceQuery}.
 	 */
-	public ImmutableMap<ExternalReferenceQuery, ExternalReference> getExternalReferences(@NonNull final Set<ExternalReferenceQuery> queries)
+	public ImmutableMap<ExternalReferenceQuery, ExternalReference> getExternalReferences(@NonNull final Collection<ExternalReferenceQuery> queries)
 	{
-		final IQueryBuilder<I_S_ExternalReference> queryBuilder = queryBL.createQueryBuilder(I_S_ExternalReference.class).setJoinOr()
+		final IQueryBuilder<I_S_ExternalReference> queryBuilder = queryBL.createQueryBuilder(I_S_ExternalReference.class)
+				.setJoinOr()
 				.setOption(IQueryBuilder.OPTION_Explode_OR_Joins_To_SQL_Unions, true);
 
 		for (final ExternalReferenceQuery query : queries)
 		{
 			queryBuilder.filter(createFilterFor(query));
 		}
 
-		final Map<ExternalReferenceQuery, ExternalReference> query2reference =
-				queryBuilder.create().stream()
-						.collect(Collectors.toMap(
-								this::buildExternalReferenceQuery, // key
-								this::buildExternalReference // value
-						));
+		final ImmutableList<ExternalReference> externalReferences = queryBuilder.create()
+				.stream()
+				.map(this::buildExternalReference)
+				.collect(ImmutableList.toImmutableList());
 
-		final ImmutableMap.Builder<ExternalReferenceQuery, ExternalReference> result = ImmutableMap.builder();
-		for (final ExternalReferenceQuery query : queries)
-		{
-			final ExternalReference externalReference = query2reference.get(query);
-			result.put(query, CoalesceUtil.coalesce(externalReference, ExternalReference.NULL));
-		}
+		final Map<ExternalReferenceQuery, ExternalReference> result = new HashMap<>();
 
-		return result.build();
+		externalReferences.forEach(externalReference -> queries.forEach(query -> {
```

Comment:
this is above my paygrade 😂 
please convert to old-school for-loops

Reply:
Converted 😄